bundles/org.openhab.binding.mqtt/src/main/java/org/openhab/binding/mqtt/discovery/AbstractMQTTDiscovery.java

```diff
@@ -36,25 +40,61 @@
  */
 @NonNullByDefault
 public abstract class AbstractMQTTDiscovery extends AbstractDiscoveryService implements MQTTTopicDiscoveryParticipant {
+
+    private final Logger logger = LoggerFactory.getLogger(AbstractMQTTDiscovery.class);
+
     protected final String subscribeTopic;
 
+    private int timeout;
+
+    private @Nullable ScheduledFuture<?> scheduledStop;
+
     public AbstractMQTTDiscovery(@Nullable Set<ThingTypeUID> supportedThingTypes, int timeout,
             boolean backgroundDiscoveryEnabledByDefault, String baseTopic) {
-        super(supportedThingTypes, timeout, backgroundDiscoveryEnabledByDefault);
+        super(supportedThingTypes, 0, backgroundDiscoveryEnabledByDefault);
         this.subscribeTopic = baseTopic;
+        this.timeout = timeout;
     }
 
     /**
      * Return the topic discovery service.
      */
     protected abstract MQTTTopicDiscoveryService getDiscoveryService();
 
+    private synchronized void stopTimeout() {
+        if (scheduledStop != null) {
```

Comment:
```suggestion
        final ScheduledFuture<?> scheduledStop = this.scheduledStop;
        if (scheduledStop != null) {
```

Reply:
What is the point?
This is a synchronized method. If it would not be, then there is a lot more to take care of....
Actually, most of the code is taken from `org.eclipse.smarthome.config.discovery.AbstractDiscoveryService`. If the code is not good, then we should refactor that class and build the resetTimeout method into that.